ydb/core/ydb_convert/ydb_convert.cpp

```diff
@@ -1464,4 +1492,96 @@ void FillOwner(NKikimrScheme::TEvModifySchemeTransaction& out,
     }
 }
 
+#define CHECK_OR_RETURN_ERROR(cond, descr) \
+    if (!(cond)) { \
+        errStr = descr; \
+        return false; \
+    }
+
+// NOTE: TCell's can reference memory from tupleValue
+bool CellsFromTuple(const Ydb::Type* tupleType,
+                    const Ydb::Value& tupleValue,
+                    const TConstArrayRef<TConversionTypeInfo>& types,
+                    bool allowCastFromString,
+                    bool checkNullability,
+                    TVector<TCell>& key,
+                    TString& errStr,
+                    TMemoryPool& valueDataPool) {
+    if (tupleType) {
+        Ydb::Type::TypeCase typeCase = tupleType->type_case();
+        CHECK_OR_RETURN_ERROR(typeCase == Ydb::Type::kTupleType ||
+                              (typeCase == Ydb::Type::TYPE_NOT_SET && tupleType->tuple_type().elementsSize() == 0), "Must be a tuple");
+        CHECK_OR_RETURN_ERROR(tupleType->tuple_type().elementsSize() <= types.size(),
+            "Tuple size " + ToString(tupleType->tuple_type().elementsSize()) + " is greater that expected size " + ToString(types.size()));
+
+        for (size_t i = 0; i < tupleType->tuple_type().elementsSize(); ++i) {
```

Comment:
I thought we use snake_case for public api protos

Reply:
It's different in different places) But in this file mostly snake case is used, so I'l fix that.

Reply:
ok